zschema/compounds.py

```diff
@@ -129,10 +156,28 @@ def print_indent_string(self, name, indent):
             value.print_indent_string(name, indent+1)
 
     def to_es(self):
-        p = {self.key_to_es(k): v.to_es() for k, v in sorted(self.definition.iteritems()) \
+        p = {self.key_to_es(k): v.to_es() \
+                for k, v in sorted(self.definition.iteritems()) \
                 if not v.exclude_elasticsearch}
         return {"properties": p}
 
+    def _docs_common(self, parent_category):
+        category = self.category if self.category else parent_category
+        retv = {
+            "category": category,
+            "doc": self.doc,
+            "type": self.__class__.__name__,
```

Comment:
It would be nice if this could be overridden...

 # `__class__.__name__` doesn't keep package information (?)
 # Constrains our type names to match python class name rules (and constrains our class names to match our type names rules?)

Beyond the scope of this PR, but I just noticed it.

Reply:
> It seems it would be nice if this could be overridden
> …
> Beyond the scope of this PR, but I just noticed it.

Good point, @justinbastress. It looks like coupling to `__class__.__name__` has been the idiom for a while (e.g., [an initial commit](https://github.com/zmap/zschema/commit/f3e1878384bceae53c3c530f8752ec6a03a8ce25#diff-6c18aa9e53ecb2eda111c7886abdd928R21) for the CLI), but an area for future improvement.